Index: ../Method/src/Order.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/Order.java b/../Method/src/Order.java
new file mode 100644
--- /dev/null	(date 1732797632527)
+++ b/../Method/src/Order.java	(date 1732797632527)
@@ -0,0 +1,17 @@
+public class Order {
+    private int count = 0;
+    private int totalCost = 0;
+
+    public void addProduct(int productCount, int cost) {
+        count = count + productCount;
+        totalCost = totalCost + cost;
+    }
+
+    public int getProductCount() {
+        return count;
+    }
+
+    public int getTotalCost() {
+        return totalCost;
+    }
+}
Index: ../Method/src/Method.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/Method.iml b/../Method/src/Method.iml
new file mode 100644
--- /dev/null	(date 1732797632531)
+++ b/../Method/src/Method.iml	(date 1732797632531)
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
Index: ../Method/src/User.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/User.java b/../Method/src/User.java
new file mode 100644
--- /dev/null	(date 1732797632521)
+++ b/../Method/src/User.java	(date 1732797632521)
@@ -0,0 +1,17 @@
+public class User {
+    public String name;
+    public int age;
+
+    User (String name, int age) {
+        this.name = name;
+        this.age = age;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getAge() {
+        return age;
+    }
+}
Index: ../InternetMagazine23102024/src/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Main.java b/../InternetMagazine23102024/src/Main.java
new file mode 100644
--- /dev/null	(date 1732797632543)
+++ b/../InternetMagazine23102024/src/Main.java	(date 1732797632543)
@@ -0,0 +1,23 @@
+public class Main {
+    public static void main(String[] args) {
+        //TODO: методы вызываемые последовательно называются стеком вызовов
+        createBasket();
+
+        Session session = new Session();
+
+        System.out.println(Basket.getCount());
+
+    }
+
+    //TODO: статический метод доступен для класса в целом (НЕ ОБЪЕКТА) и может вызываться только из другого статического метода
+    public static void createBasket() {
+        Basket basket = new Basket("goods",120);
+    }
+
+    //TODO: если предполагается работа с методом через объекты, то такой метод должен быть не статический
+    //TODO: если работа с методами предполагается за пределами объекта и работает с переменными класса, то метод должен быть статическим
+
+
+    DeliveryOrder  deliveryOrder = new DeliveryOrder(Pedestrian.PEDESTRIAN,Fragile.NO,Cold.YES);
+
+}
\ No newline at end of file
Index: ../InternetMagazine23102024/src/Basket.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Basket.java b/../InternetMagazine23102024/src/Basket.java
new file mode 100644
--- /dev/null	(date 1732797632549)
+++ b/../InternetMagazine23102024/src/Basket.java	(date 1732797632549)
@@ -0,0 +1,88 @@
+import java.time.Period;
+
+public class Basket {
+    private String items;
+    private int totalPrice = 0;
+    private int limit;
+
+    private static int count = 0; //TODO: статическая переменная доступна в самом классе, вызывается при обращении к классу.
+
+    private static int timeoute = getTimeOut();
+
+    //TODO: статическая инициализация
+    private static int getTimeOut() {
+        int secoundsInHour = 3600;
+        int hoursInDays = 24;
+        int daysInWeek  = 7;
+        int daysInMonth = 30;
+    }
+    //TODO: конструктор нужен для инциализации переменных b
+    public Basket() {
+        increaseCount(1);
+        items = "Список товаров:";
+        limit = 100000;
+    }
+
+    public Basket(int limit) {
+        this(); //TODO: вызываем конструктор пустой Basket()
+        this.limit = limit;
+    }
+
+    public Basket(String items, int totalPrice) {
+        this();
+        this.items = this.items + items;
+        this.totalPrice = totalPrice;
+    }
+
+    public static int getCount() {
+        return count;
+    }
+
+    public static void increaseCount(int count) {
+        Basket.count = Basket.count + count;
+    }
+
+    //TODO: перегрузка свойство создавать одноименные методы с разной сигнатурой
+    public void add(String name, int price) {
+        add(name, price, 1);
+    }
+
+    public void add(String name, int price, int count) {
+        if (contains(name)) {
+            return;
+        }
+        if (totalPrice + price >= limit) {
+            return;
+        }
+        items = items + "\n" + name + " - " +
+                count + " шт. - "+ price;
+        totalPrice = totalPrice + count * price;
+    }
+
+    public void add(int count, String name) {
+        add(name,500, count);
+    }
+
+    public void clear() {
+        items = "";
+        totalPrice = 0;
+    }
+
+    public int getTotalPrice() {
+        return totalPrice;
+    }
+
+    public boolean contains(String name) {
+            return items.contains(name);
+    }
+
+    public void print(String title) {
+        System.out.println(title);
+        if (items.isEmpty()) {
+            System.out.println("Корзина пуста");
+            System.out.println(items);
+        } else {
+            System.out.println(items);
+        }
+    }
+}
Index: ../Method/src/Printer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/Printer.java b/../Method/src/Printer.java
new file mode 100644
--- /dev/null	(date 1732797632537)
+++ b/../Method/src/Printer.java	(date 1732797632537)
@@ -0,0 +1,42 @@
+public class Printer {
+
+    private String queue = "";
+    private int countPrintedPages = 0;
+    private int pagesInQueue = 0;
+
+    public void append(String name, String text, int countPages) {
+        queue = queue + "\n" + name + " - " + text;
+        pagesInQueue = pagesInQueue + countPages;
+        //countPrintedPages = countPrintedPages + countPages;
+    }
+
+    public void append(String text, int countPrintPages) {
+        append("Документ без имени", text, countPrintPages);
+    }
+    public void append(String text, String name) {
+        append(name, text, 1);
+    }
+
+    public void append(String text) {
+        append("Документ без имени", text, 1);
+    }
+
+    public void clear() {
+        queue = "";
+        pagesInQueue = 0;
+    }
+
+    public void print () {
+        System.out.println(queue);
+        countPrintedPages = countPrintedPages + pagesInQueue;
+        clear();
+    }
+
+    public int getPrintedPagesCount() {
+        return countPrintedPages;
+    }
+
+    public int getPendingPagesCount() {
+        return pagesInQueue;
+    }
+}
Index: ../InternetMagazine23102024/.idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/.idea/vcs.xml b/../InternetMagazine23102024/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1732797632566)
+++ b/../InternetMagazine23102024/.idea/vcs.xml	(date 1732797632566)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$/.." vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: ../InternetMagazine23102024/.idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/.idea/modules.xml b/../InternetMagazine23102024/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1732797632575)
+++ b/../InternetMagazine23102024/.idea/modules.xml	(date 1732797632575)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/InternetMagazine23102024.iml" filepath="$PROJECT_DIR$/InternetMagazine23102024.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: ../InternetMagazine23102024/.idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/.idea/misc.xml b/../InternetMagazine23102024/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1732797632554)
+++ b/../InternetMagazine23102024/.idea/misc.xml	(date 1732797632554)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" project-jdk-name="openjdk-23" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: ../InternetMagazine23102024/.idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/.idea/.gitignore b/../InternetMagazine23102024/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632561)
+++ b/../InternetMagazine23102024/.idea/.gitignore	(date 1732797632561)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: ../NumberAndDateExperiments/NumberAndDateExperiments.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/NumberAndDateExperiments.iml b/../NumberAndDateExperiments/NumberAndDateExperiments.iml
new file mode 100644
--- /dev/null	(date 1732797632596)
+++ b/../NumberAndDateExperiments/NumberAndDateExperiments.iml	(date 1732797632596)
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
Index: ../InternetMagazine23102024/src/Cold.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Cold.java b/../InternetMagazine23102024/src/Cold.java
new file mode 100644
--- /dev/null	(date 1732797632601)
+++ b/../InternetMagazine23102024/src/Cold.java	(date 1732797632601)
@@ -0,0 +1,3 @@
+public enum Cold {
+    YES, NO
+}
Index: ../InternetMagazine23102024/src/DeliveryOrder.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/DeliveryOrder.java b/../InternetMagazine23102024/src/DeliveryOrder.java
new file mode 100644
--- /dev/null	(date 1732797632585)
+++ b/../InternetMagazine23102024/src/DeliveryOrder.java	(date 1732797632585)
@@ -0,0 +1,5 @@
+public class DeliveryOrder {
+    public DeliveryOrder(Pedestrian isPedestrian, Fragile isFragile, Cold isCold) {
+
+    }
+}
Index: ../Method/.idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/.idea/modules.xml b/../Method/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1732797632591)
+++ b/../Method/.idea/modules.xml	(date 1732797632591)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/Method.iml" filepath="$PROJECT_DIR$/Method.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: ../InternetMagazine23102024/src/Fragile.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Fragile.java b/../InternetMagazine23102024/src/Fragile.java
new file mode 100644
--- /dev/null	(date 1732797632618)
+++ b/../InternetMagazine23102024/src/Fragile.java	(date 1732797632618)
@@ -0,0 +1,3 @@
+public enum Fragile {
+    YES, NO
+}
Index: ../NumberAndDateExperiments/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/.gitignore b/../NumberAndDateExperiments/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632623)
+++ b/../NumberAndDateExperiments/.gitignore	(date 1732797632623)
@@ -0,0 +1,29 @@
+### IntelliJ IDEA ###
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+bin/
+!**/src/main/**/bin/
+!**/src/test/**/bin/
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: ../Method/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/.gitignore b/../Method/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632607)
+++ b/../Method/.gitignore	(date 1732797632607)
@@ -0,0 +1,29 @@
+### IntelliJ IDEA ###
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+bin/
+!**/src/main/**/bin/
+!**/src/test/**/bin/
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: ../Method/.idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/.idea/misc.xml b/../Method/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1732797632612)
+++ b/../Method/.idea/misc.xml	(date 1732797632612)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_19" default="true" project-jdk-name="openjdk-19" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: gitignore.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gitignore.txt b/gitignore.txt
new file mode 100644
--- /dev/null	(date 1732795812503)
+++ b/gitignore.txt	(date 1732795812503)
@@ -0,0 +1,29 @@
+### IntelliJ IDEA ###
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+bin/
+!**/src/main/**/bin/
+!**/src/test/**/bin/
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: ../Method/.idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/.idea/.gitignore b/../Method/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632637)
+++ b/../Method/.idea/.gitignore	(date 1732797632637)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: ../Method/.idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/.idea/vcs.xml b/../Method/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1732797632641)
+++ b/../Method/.idea/vcs.xml	(date 1732797632641)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$/.." vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: ../Method/Method.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/Method.iml b/../Method/Method.iml
new file mode 100644
--- /dev/null	(date 1732797632628)
+++ b/../Method/Method.iml	(date 1732797632628)
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
Index: ../InternetMagazine23102024/src/Pedestrian.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Pedestrian.java b/../InternetMagazine23102024/src/Pedestrian.java
new file mode 100644
--- /dev/null	(date 1732797632633)
+++ b/../InternetMagazine23102024/src/Pedestrian.java	(date 1732797632633)
@@ -0,0 +1,5 @@
+public enum Pedestrian {
+    PEDESTRIAN,
+    AUTO,
+    BIKE
+}
Index: .idea/gitignore.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/gitignore.txt b/.idea/gitignore.txt
new file mode 100644
--- /dev/null	(date 1731942532333)
+++ b/.idea/gitignore.txt	(date 1731942532333)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: ../InternetMagazine23102024/src/Session.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Session.java b/../InternetMagazine23102024/src/Session.java
new file mode 100644
--- /dev/null	(date 1732797632655)
+++ b/../InternetMagazine23102024/src/Session.java	(date 1732797632655)
@@ -0,0 +1,5 @@
+public class Session {
+    public Session() {
+        Basket basket = new Basket("молоко",56);
+    }
+}
Index: ../InternetMagazine23102024/src/Order.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Order.java b/../InternetMagazine23102024/src/Order.java
new file mode 100644
--- /dev/null	(date 1732797632645)
+++ b/../InternetMagazine23102024/src/Order.java	(date 1732797632645)
@@ -0,0 +1,7 @@
+public class Order {
+    public void addProduct(Product product) {
+        Product copy = new Product(product.getName(), product.getPrice());
+        //add copy to order
+
+    }
+}
Index: ../InternetMagazine23102024/src/Product.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/src/Product.java b/../InternetMagazine23102024/src/Product.java
new file mode 100644
--- /dev/null	(date 1732797632650)
+++ b/../InternetMagazine23102024/src/Product.java	(date 1732797632650)
@@ -0,0 +1,33 @@
+public class Product {
+
+    public final String name;
+    public final int price;
+
+    Product(String name, int price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    public Product setName(String name) {
+       return new Product(name, price);
+    }
+
+    public Product setPrice(int price) {
+        return new Product(name, price);
+    }
+    public String getName() {
+        return name;
+    }
+
+    public int getPrice() {
+        return price;
+    }
+
+    public String toString() {
+        return name + " - price " + price;
+    }
+    public void print() {
+        String info = name + " - price " + price;
+        System.out.println(info);
+    }
+}
Index: ../NumberAndDateExperiments/src/MoneyCalculator.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/src/MoneyCalculator.java b/../NumberAndDateExperiments/src/MoneyCalculator.java
new file mode 100644
--- /dev/null	(date 1732797632678)
+++ b/../NumberAndDateExperiments/src/MoneyCalculator.java	(date 1732797632678)
@@ -0,0 +1,13 @@
+public class MoneyCalculator {
+
+    private static final int BANKNOTES_PER_PACK = 100;
+    private static final int PACKS_IN_CASSETTE = 20;
+
+    public int calculateCassetteCount(int banknoteCount) {
+
+        int banknotePack = banknoteCount / BANKNOTES_PER_PACK + (banknoteCount % BANKNOTES_PER_PACK == 0 ? 0 :1);
+
+        int banknoteCassette = banknotePack / PACKS_IN_CASSETTE + (banknotePack % PACKS_IN_CASSETTE == 0 ? 0 : 1);
+        return banknoteCassette;
+    }
+}
Index: ../GearBox2024/src/GearBoxType.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/src/GearBoxType.java b/../GearBox2024/src/GearBoxType.java
new file mode 100644
--- /dev/null	(date 1732797632662)
+++ b/../GearBox2024/src/GearBoxType.java	(date 1732797632662)
@@ -0,0 +1,3 @@
+public enum GearBoxType {
+    AUTOMATIC, ROBOT, VARIATOR, MANUAL
+}
Index: ../GearBox2024/GearBox2024.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/GearBox2024.iml b/../GearBox2024/GearBox2024.iml
new file mode 100644
--- /dev/null	(date 1732797632668)
+++ b/../GearBox2024/GearBox2024.iml	(date 1732797632668)
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
Index: ../GearBox2024/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/.gitignore b/../GearBox2024/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632694)
+++ b/../GearBox2024/.gitignore	(date 1732797632694)
@@ -0,0 +1,29 @@
+### IntelliJ IDEA ###
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+bin/
+!**/src/main/**/bin/
+!**/src/test/**/bin/
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: ../GearBox2024/src/GearBox.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/src/GearBox.java b/../GearBox2024/src/GearBox.java
new file mode 100644
--- /dev/null	(date 1732797632699)
+++ b/../GearBox2024/src/GearBox.java	(date 1732797632699)
@@ -0,0 +1,42 @@
+public class GearBox {
+    /**
+     * 0 - neutral
+     * -1 - rear
+     * 1-6
+     */
+    //TODO: константа класса это static final и обозначаются большими буквами в нижним подчеркиванием между слов
+    //        она может быть и видимой то есть public, но окажется публичной  неизменяемой и принадлежащей классу, а не объекту класса
+
+    private int gear = 0;   //TODO: приватная переменная доступна только в самом объекте и изменяемая в самом объекте, доступная для всех методов этого класса
+    public static final int MAX_GEAR = 6;   //TODO: публичная статическая и финализируемая переменная доступная за пределами класса, является неизменяемой и статическоцй, то есть константой. Имена таких переменных обозначаются большими буквами с нижним подчёркиванием между словами
+    public static final int MIN_GEAR = 1;
+
+    public final GearBoxType type;
+
+    public GearBox(GearBoxType type) {
+        this.type = type;
+    }
+
+    public GearBoxType getType() {
+        return type;
+    }
+    public void shiftUp() {
+        gear = gear < MAX_GEAR ? gear + 1: gear;
+    }
+
+    public void shiftDown() {
+        gear = gear < MIN_GEAR ? gear - 1 : gear;
+    }
+
+    public void switchRear() {
+        gear =  (gear > 1) ? gear : - 1;
+    }
+
+    public void switchNeutral() {
+        gear = 0;
+    }
+
+    public int getCurrentGear() {
+        return gear;
+    }
+}
Index: ../NumberAndDateExperiments/src/UsageAnalytics.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/src/UsageAnalytics.java b/../NumberAndDateExperiments/src/UsageAnalytics.java
new file mode 100644
--- /dev/null	(date 1732797632684)
+++ b/../NumberAndDateExperiments/src/UsageAnalytics.java	(date 1732797632684)
@@ -0,0 +1,14 @@
+public class UsageAnalytics {
+    private int visitsCount;
+
+    public UsageAnalytics() {
+        visitsCount = 0;
+    }
+
+    public int countVisit() {
+        return ++visitsCount;
+    }
+
+
+}
+
Index: ../NumberAndDateExperiments/src/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/src/Main.java b/../NumberAndDateExperiments/src/Main.java
new file mode 100644
--- /dev/null	(date 1732797632690)
+++ b/../NumberAndDateExperiments/src/Main.java	(date 1732797632690)
@@ -0,0 +1,27 @@
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.temporal.ChronoUnit;
+
+public class Main {
+    public static void main(String[] args) {
+
+        MoneyCalculator moneyCalculator = new MoneyCalculator();
+        System.out.println(moneyCalculator.calculateCassetteCount(4001));
+
+        UsageAnalytics analytics = new UsageAnalytics();
+        analytics.countVisit();
+        analytics.countVisit();
+        int visits = analytics.countVisit();
+        System.out.println("Всего посещений: " + visits);
+
+        System.out.println(getPeriodFromJavaBirthday());
+    }
+
+    public static String getPeriodFromJavaBirthday() {
+        LocalDate dateOfBirth = LocalDate.of(1995,5,23);
+        Period period = dateOfBirth.until(LocalDate.now());
+
+        return period.getYears() + " years, " + period.getMonths() + " months, " +
+                period.getDays() + " days";
+    }
+}
\ No newline at end of file
Index: ../GearBox2024/.idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/.idea/modules.xml b/../GearBox2024/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1732797632713)
+++ b/../GearBox2024/.idea/modules.xml	(date 1732797632713)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/GearBox2024.iml" filepath="$PROJECT_DIR$/GearBox2024.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: ../GearBox2024/.idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/.idea/misc.xml b/../GearBox2024/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1732797632718)
+++ b/../GearBox2024/.idea/misc.xml	(date 1732797632718)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_22" default="true" project-jdk-name="22" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: ../GearBox2024/src/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/src/Main.java b/../GearBox2024/src/Main.java
new file mode 100644
--- /dev/null	(date 1732797632704)
+++ b/../GearBox2024/src/Main.java	(date 1732797632704)
@@ -0,0 +1,8 @@
+public class Main {
+    public static void main(String[] args) {
+        GearBoxType type = GearBoxType.valueOf("ROBOT");
+        GearBox gearBox = new GearBox(type);
+        System.out.println(GearBox.MAX_GEAR);
+
+        System.out.println(gearBox.getType());    }
+}
\ No newline at end of file
Index: ../GearBox2024/.idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/.idea/.gitignore b/../GearBox2024/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632709)
+++ b/../GearBox2024/.idea/.gitignore	(date 1732797632709)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: ../GearBox2024/.idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../GearBox2024/.idea/vcs.xml b/../GearBox2024/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1732797632727)
+++ b/../GearBox2024/.idea/vcs.xml	(date 1732797632727)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$/.." vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: ../Method/src/.idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/.idea/modules.xml b/../Method/src/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1732797632732)
+++ b/../Method/src/.idea/modules.xml	(date 1732797632732)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/Method.iml" filepath="$PROJECT_DIR$/Method.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: ../Method/src/.idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/.idea/vcs.xml b/../Method/src/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1732797632722)
+++ b/../Method/src/.idea/vcs.xml	(date 1732797632722)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$/../.." vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: ../Method/src/.idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/.idea/.gitignore b/../Method/src/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632745)
+++ b/../Method/src/.idea/.gitignore	(date 1732797632745)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: ../NumberAndDateExperiments/.idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/.idea/modules.xml b/../NumberAndDateExperiments/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1732797632750)
+++ b/../NumberAndDateExperiments/.idea/modules.xml	(date 1732797632750)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/NumberAndDateExperiments.iml" filepath="$PROJECT_DIR$/NumberAndDateExperiments.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: ../NumberAndDateExperiments/.idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/.idea/.gitignore b/../NumberAndDateExperiments/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632736)
+++ b/../NumberAndDateExperiments/.idea/.gitignore	(date 1732797632736)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: ../Method/src/.idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/.idea/misc.xml b/../Method/src/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1732797632741)
+++ b/../Method/src/.idea/misc.xml	(date 1732797632741)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_22" default="true" project-jdk-name="22" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: .idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/.gitignore b/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1732795812438)
+++ b/.idea/.gitignore	(date 1732795812438)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: ../NumberAndDateExperiments/.idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/.idea/vcs.xml b/../NumberAndDateExperiments/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1732797632765)
+++ b/../NumberAndDateExperiments/.idea/vcs.xml	(date 1732797632765)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$/.." vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: ../InternetMagazine23102024/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/.gitignore b/../InternetMagazine23102024/.gitignore
new file mode 100644
--- /dev/null	(date 1732797632772)
+++ b/../InternetMagazine23102024/.gitignore	(date 1732797632772)
@@ -0,0 +1,29 @@
+### IntelliJ IDEA ###
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+bin/
+!**/src/main/**/bin/
+!**/src/test/**/bin/
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: ../Method/gitignore.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/gitignore.txt b/../Method/gitignore.txt
new file mode 100644
--- /dev/null	(date 1732797632754)
+++ b/../Method/gitignore.txt	(date 1732797632754)
@@ -0,0 +1,29 @@
+### IntelliJ IDEA ###
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+bin/
+!**/src/main/**/bin/
+!**/src/test/**/bin/
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: ../Method/src/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../Method/src/Main.java b/../Method/src/Main.java
new file mode 100644
--- /dev/null	(date 1732797632760)
+++ b/../Method/src/Main.java	(date 1732797632760)
@@ -0,0 +1,21 @@
+import javax.swing.undo.UndoableEditSupport;
+import java.time.LocalDateTime;
+import java.util.Formatter;
+import java.util.Scanner;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Printer printer = new Printer();
+        printer.append("some text 133", "Справка");
+        printer.append("some text 210", 544);
+        System.out.println(printer.getPendingPagesCount());
+        printer.print();
+        printer.append("some text 8",530);
+        printer.clear();
+        printer.append("some text 23","Договор",312);
+        printer.append("some text 11");
+        printer.print();
+        System.out.println(printer.getPrintedPagesCount());
+    }
+}
\ No newline at end of file
Index: ../NumberAndDateExperiments/.idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../NumberAndDateExperiments/.idea/misc.xml b/../NumberAndDateExperiments/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1732797632778)
+++ b/../NumberAndDateExperiments/.idea/misc.xml	(date 1732797632778)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_22" default="true" project-jdk-name="22" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: src/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Main.java b/src/Main.java
new file mode 100644
--- /dev/null	(date 1732795870042)
+++ b/src/Main.java	(date 1732795870042)
@@ -0,0 +1,70 @@
+import javax.crypto.spec.PSource;
+
+public class Main {
+    //TODO: заполнение примитивов значениями по умолчанию происходит только,
+    // когда примитив является параметром класса, как ниже:
+    public static Product product;
+    public static int price;
+    public static void main(String[] args) {
+        //TODO: в методе при объявлении переменной ей обязательно нужно присвоить значение - инициализировать
+        int milkPrice;
+        System.out.println(product);
+        System.out.println(price);
+        //System.out.println(milkPrice);
+        byte b = 120;
+        //byte v = b + b; //*переполнение
+        long s = 4643213245454654641L;
+
+        /*
+        /7.4. BYTE - Байт
+        /систем счисления
+        /десятеричная цифры  0-9
+        /двоичная, цифры 0-1,
+        /шестнадцатеричная, цифры 0-9, буквы A-F
+        /всего 256 символов от -128 до 127
+        /Mbps - Megabits per second
+        /Биты кило, мега - 1000, 100000
+        /Байты: 1 Кб = 1024, 1 Мб = 1024*1024 = 1048576
+         */
+
+        /*
+        7.5 Числа (типы)
+        byte - 8 бит (от -128 до 127)
+        short - 16 бит (-32768 до 32767)
+        int   - 32 бит (-2147483648 до 2147483647)
+        long  -  64 бита.
+         */
+
+        byte d = 127;
+        d++;
+        System.out.println(d);
+
+        int a = 2000000000;
+        int k = -1500000000;
+        int comparison = a - k;
+        System.out.println(comparison);;
+        if (comparison > 0) {
+            System.out.println("Число a больше числа k");
+        } else {
+            System.out.println("Число k больше числа а");
+        }
+
+        //числа с плавающей точкой
+        //float и double, занимают 32 и 64 байта в памяти
+
+        double n = -820645000000000.0;
+        System.out.println(n);
+
+        float l = 140000000000000000000F;
+        System.out.println(l);
+
+
+
+
+
+    }
+
+    public static void addComission(Product product) {
+        product.setPrice(product.getPrice() + 10);
+    }
+}
\ No newline at end of file
Index: .gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.gitignore b/.gitignore
new file mode 100644
--- /dev/null	(date 1731050931121)
+++ b/.gitignore	(date 1731050931121)
@@ -0,0 +1,29 @@
+### IntelliJ IDEA ###
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+bin/
+!**/src/main/**/bin/
+!**/src/test/**/bin/
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: PrimitivesAndObjects.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/PrimitivesAndObjects.iml b/PrimitivesAndObjects.iml
new file mode 100644
--- /dev/null	(date 1731050931857)
+++ b/PrimitivesAndObjects.iml	(date 1731050931857)
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
Index: src/Product.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Product.java b/src/Product.java
new file mode 100644
--- /dev/null	(date 1731051464934)
+++ b/src/Product.java	(date 1731051464934)
@@ -0,0 +1,29 @@
+public class Product {
+    private String name;
+    private int price;
+
+    public Product(String name, int price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    public void setPrice(int price) {
+        this.price = price;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getPrice() {
+        return price;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String toString() {
+        return name + " - " + price;
+    }
+}
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1732795812439)
+++ b/.idea/misc.xml	(date 1732795812439)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_22" default="true" project-jdk-name="22" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: src/Person.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Person.java b/src/Person.java
new file mode 100644
--- /dev/null	(date 1732795812504)
+++ b/src/Person.java	(date 1732795812504)
@@ -0,0 +1,17 @@
+public class Person {
+
+    public String name;
+
+    public int birthYear;
+
+
+
+    public Person(String name, int birthYear) {
+
+        this.name = name;
+
+        this.birthYear = birthYear;
+
+    }
+
+}
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/vcs.xml b/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1732795812439)
+++ b/.idea/vcs.xml	(date 1732795812439)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$/.." vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/modules.xml b/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1732795812439)
+++ b/.idea/modules.xml	(date 1732795812439)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/PrimitivesAndObjects.iml" filepath="$PROJECT_DIR$/PrimitivesAndObjects.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: ../InternetMagazine23102024/InternetMagazine23102024.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../InternetMagazine23102024/InternetMagazine23102024.iml b/../InternetMagazine23102024/InternetMagazine23102024.iml
new file mode 100644
--- /dev/null	(date 1732797632515)
+++ b/../InternetMagazine23102024/InternetMagazine23102024.iml	(date 1732797632515)
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
